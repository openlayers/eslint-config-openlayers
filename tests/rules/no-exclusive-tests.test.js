import {createRuleTester} from 'eslint-vitest-rule-tester';
import {describe, expect, test} from 'vitest';
import configs from '../../index.js';
import rule from '../../rules/no-exclusive-tests.js';

describe('no-exclusive-tests', () => {
  const {valid, invalid} = createRuleTester({
    name: 'no-exclusive-tests',
    rule,
    configs: {
      languageOptions: configs[0].languageOptions,
    },
  });

  test('valid use of "describe()" and "it()"', () => {
    valid({
      filename: 'valid.test.js',
      code: `
        describe('some functionality', () => {
          it('works', () => {
            alert('woohoo');
          });
        });
      `,
    });
  });

  test('valid use of "describe.only.notACallExpression"', () => {
    valid({
      filename: 'valid.test.js',
      code: `
        describe.only.notACallExpression = 'ok';
      `,
    });
  });

  test('invalid use of "describe.only()"', () => {
    const result = invalid({
      filename: 'invalid.test.js',
      code: `
        describe.only('example invalid use of "describe.only()"', () => {
          it('example valid use of "it()"', () => {
            alert('invalid');
          });
        });
      `,
    });

    expect(result.messages).toMatchSnapshot();
  });

  test('invalid use of "it.only()"', () => {
    const result = invalid({
      filename: 'invalid.test.js',
      code: `
        describe('example valid use of "describe()"', () => {
          it.only('example invalid use of "it.only()"', () => {
            alert('invalid');
          });
        });
      `,
    });

    expect(result.messages).toMatchSnapshot();
  });

  test('invalid use of "describe.only()" and "it.only()"', () => {
    const result = invalid({
      filename: 'invalid.test.js',
      code: `
        describe.only('example invalid use of "describe.only()"', () => {
          it.only('example invalid use of "it.only()"', () => {
            alert('invalid');
          });
        });
      `,
    });

    expect(result.messages).toMatchSnapshot();
  });
});
