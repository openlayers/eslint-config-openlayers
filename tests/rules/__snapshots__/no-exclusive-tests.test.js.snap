// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`no-exclusive-tests > invalid use of "describe.only()" 1`] = `
[
  {
    "column": 18,
    "endColumn": 22,
    "endLine": 2,
    "line": 2,
    "message": "Exlusive tests are not allowed",
    "nodeType": "Identifier",
    "ruleId": "no-exclusive-tests",
    "severity": 2,
  },
]
`;

exports[`no-exclusive-tests > invalid use of "describe.only()" and "it.only()" 1`] = `
[
  {
    "column": 18,
    "endColumn": 22,
    "endLine": 2,
    "line": 2,
    "message": "Exlusive tests are not allowed",
    "nodeType": "Identifier",
    "ruleId": "no-exclusive-tests",
    "severity": 2,
  },
  {
    "column": 14,
    "endColumn": 18,
    "endLine": 3,
    "line": 3,
    "message": "Exlusive tests are not allowed",
    "nodeType": "Identifier",
    "ruleId": "no-exclusive-tests",
    "severity": 2,
  },
]
`;

exports[`no-exclusive-tests > invalid use of "it.only()" 1`] = `
[
  {
    "column": 14,
    "endColumn": 18,
    "endLine": 3,
    "line": 3,
    "message": "Exlusive tests are not allowed",
    "nodeType": "Identifier",
    "ruleId": "no-exclusive-tests",
    "severity": 2,
  },
]
`;
